### ISSUE TITLE
Weapon templates, saves, and other properties are not set

### ISSUE THREAD

**Issue by gravagehulk** [unrealkakeman89/sw5e#217]
First of all, I want to thank you for all of your hard work on this. I have always wanted to be able to run a Star Wars game with 5e rules and this system is pretty complete and functional. 

As I have been testing, I noticed that most of the weapons, force powers, etc do not work correctly. Looking at the properties of each item, it seems like the template settings, saving throws, and other misc settings are missing from most of them. I have been coding them myself, but I had to import all of the "monsters" to items to do that and I just wanted to make you aware if it's not on your radar. This is even more important if someone is using Midi-QoL, as it needs this information to apply effects and damage properly.

I have also reached out to the developer of automated animations. He was able to quickly update his module to support SW5e. Now we can have blaster lasers flying across the screen with accompanying sounds. It works marvelously, though you have to set the sound/animation for each weapon because the module was written for 5e and doesn't know what a blaster pistol or lightsaber is.

One last question: Are you planning to have a ships compendium? I created a few of my own using your rules for the time being, but it seems weird not having Tie Fighters and X-Wings in a Star Wars game. 

Thank you!

**Comment by ExileofBrokenSky** [comment]
Ship sheets are actively being worked on and are coming Soon™ as is 0.8.x compatibility. 
In regards to powers, we'll have to give them another round of polish.

**Comment by gravagehulk** [comment]
> Ship sheets are actively being worked on and are coming Soon™ as is 0.8.x compatibility.
> In regards to powers, we'll have to give them another round of polish.

Thanks for the reply! For what it's worth, I've been manually editing each Monster and going through all of the abilities and it seems to be mostly abilities with saving throws, templates (grenades, flame throwers, etc, and damage types not being selected.

**Comment by unrealkakeman89** [comment]
@gravagehulk once you have updated the Monsters, would you mind exporting and sending?

**Comment by gravagehulk** [comment]
> @gravagehulk once you have updated the Monsters, would you mind exporting and sending?

I don't have a problem doing this to help out, but there would be a few issues with my set:

1) I also code them to work with Automated Active Effects (for blaster fire, lightsabers, etc) to add sounds and animations. Those will be included in what I export.

2) There were several "monsters" that had abilities that seemed to be copied over from other "monsters" to the point that the description was mentioning the monster they were copied from. I wasn't honestly sure what to do with those, so I just made them work properly and called it a day. 

I haven't even really started on the Force powers for the force users or the technology powers. I concentrated on most of the weapon attacks.

**Comment by supervj** [comment]
When we had the powers update a while back we went through all the monsters and gave them their proper foce and tech powers, so those should be fine.

**Comment by gravagehulk** [comment]
> When we had the powers update a while back we went through all the monsters and gave them their proper foce and tech powers, so those should be fine.

Okay, sounds good. I'll finish fixing the attacks and then export. Do you have a preferred format? Or should I just use adventure exporter?

**Comment by supervj** [comment]
Are you updating them in the compendium directly? If so you can then just submit a pull request for the monsters.db file in the packs directory

**Comment by gravagehulk** [comment]
Well, I knew they would get overridden with the update, so I imported them to actors/items, then updated them, then exported to my own compendium. There are still some monsters I need to update, but I have been using them that way for my game for a few weeks.

**Comment by supervj** [comment]
Okay so you can just submit the new compendium and we can do the integration. We will likely have to fix the ids to match our compendium.

**Comment by gravagehulk** [comment]
Just an update: I am working on this. I'm on the "D's" currently. I run three games/week and work full time, so I've had to cram it in when I can. I originally coded just the monsters I knew I'd be using in my campaign, so I have a lot more to do.

I do run across some confusing creatures from time to time, though. Like the General Grievous one that is supposed to have legendary hit point pools. There are no rules for that, nor a mechanic or even a mention of how they're supposed to work.

That's just an example of several I've encountered. I'm doing my best just to work around it and fix the standard actions on each one. I'll post the first draft when I'm done, hopefully within a few weeks.

**Comment by gravagehulk** [comment]
Another update: I am almost done with this. There was more to fix than I expected  and some really confusing stuff, too (all of the Mandalorian NPCs had bite, tail, and thunderstorm attacks, for some reason).

I'm on the letter "S". Should be able to power through S-Z this week and then I'll export it for you. It's just a first pass. I'm sure there are things I missed.

**Comment by gravagehulk** [comment]
Alright, guys. here's the first draft. I had a lot more to fix than I anticipated. Apologies for the delay. I consider this to be a first pass. I know there are things I missed. I plan to go through all of them with a fine tooth comb again. There were also some abilities that had text from other monsters, so I just left them. I wasn't sure if it was a mistake or if the text just wasn't clean up for that record. Most of the saving throws, templates, etc are set now. I also set the recharge abilities, legendary actions that had limited uses, etc. 
[monsters-fixed.db.zip](https://github.com/unrealkakeman89/sw5e/files/7199433/monsters-fixed.db.zip)
